shapash/explainer/smart_plotter.py

```diff
@@ -2323,3 +2332,430 @@ def correlations(
+        max_features : int, optional
+            Maximum number of displayed features, by default 10
+        file_name: string, optional
+            Specify the save path of html files. If it is not provided, no file will be saved, by default None
+        auto_open: bool, optional
+            open automatically the plot, by default False
+
+        Returns
+        -------
+        fig
+            The figure that will be displayed
+        """        
+        assert index in self.explainer.x_pred.index, "index must exist in pandas dataframe"
+
+        self.explainer.compute_features_stability([index])
+
+        column_names = np.array([self.explainer.features_dict.get(x) for x in self.explainer.x_pred.columns])
+        ordinal = lambda n: "%d%s" % (n, "tsnrhtdd"[(math.floor(n / 10) % 10 != 1) * (n % 10 < 4) * n % 10:: 4])
+
+        # Compute explanations for instance and neighbors
+        g = self.explainer.local_neighbors["norm_shap"]
+
+        # Reorder indices based on absolute values of the 1st row (i.e. the instance) in descending order
+        inds = np.flip(np.abs(g[0, :]).argsort())
+        g = g[:, inds]
+        columns = [column_names[i] for i in inds]
+
+        # Plot
+        g_df = pd.DataFrame(g, columns=columns).T.rename(
+                columns={
+                    **{0: "instance", 1: "closest neighbor"},
+                    **{i: ordinal(i) + " closest neighbor" for i in range(2, len(g))},
+                }
+            )
+
+        # Keep only max_features
+        if max_features is not None: g_df = g_df[:max_features]
+
+        fig = go.Figure(data=[go.Bar(name=g_df.iloc[::-1, ::-1].columns[i],
+                        y=g_df.iloc[::-1, ::-1].index.tolist(),
+                        x=g_df.iloc[::-1, ::-1].iloc[:, i],
+                        marker=dict(color=self.dict_compare_colors[:-1]),
+                        orientation='h') for i in range(g_df.shape[1])])
+
+        fig.update_layout(title={'text': "Explanation of local stability:\
+                How similar are explanations for closeby neighbours?"},
+                          xaxis_title="Normalized SHAP value",
+                          barmode="group",
+                          bargap=0.5,
+                          height=30*g_df.shape[0]*g_df.shape[1] if g_df.shape[0] < 100 else 10*g_df.shape[0]*g_df.shape[1],
+                          width=900,
+                          legend={"traceorder": "reversed"},
+                          xaxis={"side": "top"},
+                          margin=dict(t=185))
+
+        if file_name is not None:
+            plot(fig, filename=file_name, auto_open=auto_open)
+
+        return fig
+
+    def stability_plot(self, selection=None, max_points=500, force=False, max_features=10, distribution='none', file_name=None, auto_open=False):
+        """The idea behind local stability is the following : if instances are very similar, then one would expect the explanations to be similar as well.
```

Comment:
we should add a sentence that presents the main objective: "The Stability_plot has the main objective..."